cli/flox-rust-sdk/src/providers/buildenv.rs

```diff
@@ -196,38 +517,683 @@ impl BuildEnv for BuildEnvNix {
+            let requisites_path = out_path.join("requisites.txt");
+            assert!(requisites_path.exists());
+
+            let requisites: HashSet<String> = std::fs::read_to_string(&requisites_path)
+                .unwrap()
+                .lines()
+                .map(String::from)
+                .collect();
+
+            let output = Command::new("nix-store")
+                .arg("-qR")
+                .arg(out_path)
+                .output()
+                .expect("failed to execute process");
+
+            assert!(output.status.success());
+
+            let store_paths: HashSet<String> = String::from_utf8_lossy(&output.stdout)
+                .lines()
+                .map(String::from)
+                .collect();
+
+            assert_eq!(requisites, store_paths);
+        }
+    }
+
+    #[test]
+    fn detects_conflicting_packages() {
+        let buildenv = BuildEnvNix;
+        let lockfile_path = GENERATED_DATA.join("envs/vim-vim-full-conflict.json");
+        let result = buildenv.build(&lockfile_path, None);
+        let err = result.expect_err("conflicting packages should fail to build");
+
+        let BuildEnvError::Build(output) = err else {
+            panic!("expected build to fail, got {}", err);
+        };
+
+        let output_matches = Regex::new("error: collision between .*-vim-.* and .*-vim-.*")
+            .unwrap()
+            .is_match(&output);
+
+        assert!(
+            output_matches,
+            "expected output to contain a conflict message: {output}"
+        );
+    }
+
+    #[test]
+    fn resolves_conflicting_packages_with_priority() {
+        let buildenv = BuildEnvNix;
+        let lockfile_path = GENERATED_DATA.join("envs/vim-vim-full-conflict-resolved.json");
+        let result = buildenv.build(&lockfile_path, None);
+        assert!(
+            result.is_ok(),
+            "conflicting packages should be resolved by priority"
+        );
+    }
+
+    /// Single quotes in variables should be escaped.
+    /// Similarly accidentally escaped single quotes like
+    ///
+    /// ```text
+    /// [vars]
+    /// singlequoteescaped = "\\'baz"
+    /// ```
+    /// should be escaped and printed as   `\'baz` (literally)
+    #[test]
+    fn environment_escapes_variables() {
+        let buildenv = BuildEnvNix;
+        let lockfile_path = MANUALLY_GENERATED.join("buildenv/lockfiles/vars_escape/manifest.lock");
+        let result = buildenv.build(&lockfile_path, None).unwrap();
+
+        let runtime = result.runtime.as_ref();
+        let develop = result.develop.as_ref();
+
+        for envrc_path in [
+            runtime.join("activate.d/envrc"),
+            develop.join("activate.d/envrc"),
+        ] {
+            assert!(envrc_path.exists());
+            let content = std::fs::read_to_string(&envrc_path).unwrap();
+            assert!(content.contains(r#"export singlequotes="'bar'""#));
+            assert!(content.contains(r#"export singlequoteescaped="\'baz""#));
```

Comment:
(nb, praise) That's easier to read than: https://github.com/flox/flox/blob/6916afa816b6b59725a735f59f70b7cb643036f0/cli/tests/buildenv.bats#L152-L153